### PR TITLE
Feature/angular components directives pipes

### PR DESCRIPTION
**Pull request by AnatoliRub**
### Task scope:
Once a user is entered the app, The only Header component should be shown. For now, you can enter whatever value which won't influence your search result. Please note that the Sorting criteria block should be hidden
By submitting the search form, the Search results block appears.
Using the response example, render cards with predefined data You can store the mocked data in the Search results component

The bottom border represents the publication date status. Create a directive to achieve the result:
* If a publication date is less than a month, set border background to green
* If a publication date is less than 7 days, set border background to blue
* If a publication date is more than 6 months, set border background to red

By clicking on the Settings button, the Filtering criteria block should be toggled
It should be possible to sort search results by date or count of views
Sorting should work both in the direction of decreasing values and in the direction of increasing values
Using a pipe, filter search results by value that a user types in the input

### CODE REVIEW

File: youtube-client/src/app/app.component.ts [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
-  title = 'youtube-client';
+  isShowSettings: boolean = false;
+
+  isShowContent: boolean = false;
+
+  searchString: string = '';
+
+  filterData: IFilterData = { order: Order.Desc, filterType: Filter.Default };
+
+  showSettings() {
```
Comment (StanevichIvan) [inline youtube-client/src/app/app.component.ts:17]: Might `toggleSettings` it's show and hide settings as well

File: youtube-client/src/app/app.component.ts [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
+    this.isShowSettings = !this.isShowSettings;
+  }
+
+  showResults(value: string) {
+    if (!this.isShowContent) {
```
Comment (StanevichIvan) [inline youtube-client/src/app/app.component.ts:22]: Might we could simplify that with ternary operator?

File: youtube-client/src/app/pipes/filter-data-by-criteria.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Filter, Order } from 'src/types/filtering-criteria-types';
+import { Post } from 'src/types/youtube-data';
+
+@Pipe({
+  name: 'filterDataByCriteria',
+})
+export class FilterDataByCriteriaPipe implements PipeTransform {
+  transform = (posts: Post[], type: Filter, order?: string, word?: string): Post[] => {
+    switch (type) {
```
Comment (StanevichIvan) [inline youtube-client/src/app/pipes/filter-data-by-criteria.pipe.ts:10]: might if with early returns suits a bit better for that?

File: youtube-client/src/app/components/search-result/card/card.component.ts [hunk 4]
```diff
@@ -7,15 +8,68 @@ import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
   styleUrls: ['./card.component.scss'],
 })
 export class CardComponent {
-  @Input() views: string = '';
+  @Input() post: Post = {
```
Comment (StanevichIvan) [inline youtube-client/src/app/components/search-result/card/card.component.ts:11]: What the reason of this fake Post?

File: youtube-client/src/app/app.component.ts (hunk 1 above)
Comment (AnatoliRub) [reply youtube-client/src/app/app.component.ts]: 581dd1a

File: youtube-client/src/app/app.component.ts (hunk 2 above)
Comment (AnatoliRub) [reply youtube-client/src/app/app.component.ts]: 581dd1a

File: youtube-client/src/app/pipes/filter-data-by-criteria.pipe.ts (hunk 3 above)
Comment (AnatoliRub) [reply youtube-client/src/app/pipes/filter-data-by-criteria.pipe.ts]: 0474c7f

File: youtube-client/src/app/components/search-result/card/card.component.ts (hunk 4 above)
Comment (AnatoliRub) [reply youtube-client/src/app/components/search-result/card/card.component.ts]: b4a4e00

**Review by StanevichIvan — APPROVED** [review]

**Pull request merged**